benchmark/crypto/hash.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+#include <algorithm>
+#include <crypto/hash.hpp>
+
+
+static void HASH_Sha3_256_with_keccak(benchmark::State& state) {
+  std::string s = "0123456789";  // length = 10
```

Comment:
It is just to assess the relative speed of different hash implementations.

For example, there is `static inline digest_to_hexdigest` function. Whithout inline it works 15% longer.